devel/fatal/Portfile

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim: fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        facebook fatal 2022.12.26.00 v
+revision            0
+categories          devel
+license             BSD
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         Library for fast prototyping software in modern C++
+long_description    Fatal is a library for fast prototyping software in modern C++. \
+                    It provides facilities to enhance the expressive power of C++. \
+                    The library is heavily based on template meta-programming, \
+                    while keeping the complexity under-the-hood.
+checksums           rmd160  cea1204d92930efee0da93c1444ebf53e60c6471 \
+                    sha256  40b7c4a644ad663232d255fffbdfd87be09872304165c67f46cc93c052319532 \
+                    size    651285
+github.tarball_from releases
+distname            ${name}-v${version}
+extract.mkdir       yes
+
+installs_libs       no
+supported_archs noarch
+
+
+use_configure       no
+build               {}
+
+destroot {
+    copy ${worksrcpath}/${name} ${destroot}${prefix}/include
+    xinstall -d ${destroot}${prefix}/share/${name}
+    xinstall -W ${worksrcpath} README.md CONTRIBUTING.md LICENSE PATENTS \
+        ${destroot}${prefix}/share/${name}
+}
+
+post-destroot {
+    fs-traverse f ${destroot}${prefix}/include/${name} {
+        if {[file isfile ${f}] && [file extension ${f}] == ".cpp"} {
+            delete ${f}
+        }
+    }
+}
```

Comment:
not sure I understand the reason behind this - why do you want to remove these files? If that's needed then it would probably be better to use "glob".

Reply:
It is a header-only library, we do not need .cpp test sources to be installed, I believe.

Reply:
so why don't you just only copy the header files you want in the "destroot" phase?

Reply:
Is there an easy way to do that, preserving directories structure? Headers are not just in one include folder, there are multiple directories inside, and .cpp files are in subdirectories there.
There is a way perhaps, I just do not know it.

Reply:
well, I don’t know from the top of my head whether “glob” will handle that - you’d need to try/check. 

I guess one could use the “traverse-fs” in the destroot ohse and copy everything that does not have the extension .cpp”.

Reply:
TBH, I do not think we will have significantly less code in result. Is it a matter of aesthetics or do I miss something? Result gonna be strictly identical, even if I write an alternative implementation.

Reply:
@reneeotten I looked again at this, and I think it is optimal as-is. There is a complex structure of folders, we do not want to throw all headers into one jar. Deleting `.cpp` leaves headers at their respective places, and the whole structure is moved to destroot. I can’t think of an easy way to pick headers instead, and don’t really get why that is any better.